RadixWallet/Features/AccountHistory/TransactionHistoryFilters+View.swift

```diff
@@ -0,0 +1,298 @@
+						}
+
+						HStack(spacing: .zero) {
+							FlowLayout(spacing: spacing) {
+								FiltersView(filters: filters, store: store)
+							}
+							.measureSize(flowLayoutID)
+							.overlay {
+								TransactionFilterView.Dummy()
+									.measureSize(flowDummyID)
+							}
+
+							Spacer(minLength: 0)
+						}
+						.frame(maxHeight: isCollapsible && isCollapsed ? collapsedHeight : .infinity, alignment: .top)
+						.clipped()
+						.onReadSizes(flowDummyID, flowLayoutID) { dummySize, flowSize in
+							if isCollapsible {
+								rowHeight = dummySize.height
+								totalHeight = flowSize.height
+							}
+						}
+
+						if totalHeight > collapsedHeight, let labels {
+							Button {
+								withAnimation {
+									isCollapsed.toggle()
+								}
+							} label: {
+								ZStack {
+									Text("+ \(labels.showAll)")
+										.opacity(isCollapsed ? 1 : 0)
+									Text("- \(labels.showLess)")
+										.opacity(isCollapsed ? 0 : 1)
+								}
+							}
+							.buttonStyle(.blueText)
+							.padding(.top, .medium3)
+						}
+					}
+					.animation(.default, value: isCollapsed)
+				}
+			}
+
+			private let flowLayoutID = "FlowLayout"
+			private let flowDummyID = "FlowDummy"
+		}
+
+		private struct FiltersView: SwiftUI.View {
+			let filters: IdentifiedArrayOf<State.Filter>
+			let store: StoreOf<TransactionHistoryFilters>
+
+			var body: some SwiftUI.View {
+				ForEach(filters) { filter in
+					TransactionFilterView(filter: filter) { id in
+						store.send(.view(.filterTapped(id)))
+					}
+				}
+			}
+		}
+	}
+}
+
+// MARK: - TransactionFilterView
+struct TransactionFilterView: SwiftUI.View {
+	let filter: TransactionHistoryFilters.State.Filter
+	let action: (TransactionFilter) -> Void
+	var crossAction: ((TransactionFilter) -> Void)? = nil
+
+	var body: some SwiftUI.View {
+		Button {
+			action(filter.id)
+		} label: {
+			HStack(spacing: .small3) {
+				if let icon = filter.icon {
+					Image(icon)
+				}
+
+				Text(filter.label)
+					.textStyle(.body1HighImportance)
+					.foregroundStyle(textColor)
+			}
+			.padding(.vertical, .small2)
+			.padding(.horizontal, .medium3)
+		}
+		.contentShape(Capsule())
+		.disabled(showCross)
+		.padding(.trailing, showCross ? .medium1 : 0)
+		.background {
+			ZStack {
+				Capsule().fill(filter.isActive ? .app.gray1 : .app.white)
+				Capsule().stroke(filter.isActive ? .clear : .app.gray3)
+			}
+		}
+		.overlay(alignment: .trailing) {
+			if showCross, let crossAction {
+				Button(asset: AssetResource.close) {
+					crossAction(filter.id)
+				}
+				.tint(.app.gray3)
+				.padding(.vertical, -.small3)
+				.padding(.trailing, .small2)
+				.transition(.scale.combined(with: .opacity))
+			}
+		}
+		.animation(.default.speed(2), value: filter.isActive)
+	}
+
+	private var showCross: Bool {
+		crossAction != nil && filter.isActive
+	}
+
+	private var textColor: Color {
+		filter.isActive ? .app.white : .app.gray1
+	}
+
+	struct Dummy: SwiftUI.View {
```

Comment:
Dummy needed still?

Reply:
Yeah this one is still needed, I didn't manage to get the FlowLayout itself handle this, I couldn't see how that could possibly be animated properly.